orca-core/src/test/java/com/netflix/spinnaker/orca/pipeline/expressions/ExpressionTransformTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2019 Pivotal, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.pipeline.expressions;
+
+import com.google.common.collect.ImmutableSet;
+import com.netflix.spinnaker.orca.pipeline.model.Execution;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+
+import static java.util.Collections.singletonList;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ExpressionTransformTest {
+  private final Set<ExpressionFunctionProvider> expressionFunctionProviders
+    = new ImmutableSet.Builder<ExpressionFunctionProvider>()
+    .add(new TestExpressionFunctionProviderWithNamespace())
+    .add(new TestExpressionFunctionProviderWithoutNamespace())
+    .build();
+  private final ExpressionTransform expressionTransform = new ExpressionTransform(expressionFunctionProviders, null, null);
+
+  @Test
+  void includeExecutionParameterAccountsForNamespaceInExpressionFunctionProvider() {
+    String expression = "${#testNamespace.testFunctionInNamespace(\"param\")}";
+    String expectedResult = "${#testNamespace_testFunctionInNamespace( #root.execution, \"param\")}";
+
+    String result = expressionTransform.includeExecutionParameter(expression);
+
+    assertThat(result).isEqualTo(expectedResult);
+  }
+
+  @Test
+  void includeExecutionParameterDoesNotInsertNullNamespaceInExpressionFunctionProvider() {
+    String expression = "${#testFunctionNoNamespace(\"param\")}";
+    String expectedResult = "${#testFunctionNoNamespace( #root.execution, \"param\")}";
+
+    String result = expressionTransform.includeExecutionParameter(expression);
+
+    assertThat(result).isEqualTo(expectedResult);
+  }
+
+  private static class TestExpressionFunctionProviderWithNamespace implements ExpressionFunctionProvider {
+    @Nullable
+    @Override
+    public String getNamespace() {
+      return "testNamespace";
+    }
+
+    @NotNull
+    @Override
+    public Collection<FunctionDefinition> getFunctions() {
+      return Collections.singletonList(
+        new FunctionDefinition("testFunctionInNamespace", singletonList(
+          new FunctionParameter(Execution.class, "functionParam", "Function param")
```

Comment:
from the usage of how you have it above, this should be:
```
Arrays.asList(
        new FunctionParameter(Execution.class, "execution", "Execution param"),
        new FunctionParameter(String.class, "functionParam", "Function param")
))
```

instead of `singletonList`

Reply:
👍